src/data_loader.cc

```diff
@@ -451,6 +451,48 @@ DataLoader::ValidateIndexes(int stream_id, int step_id) const
   return cb::Error::Success;
 }
 
+size_t
+DataLoader::GetDatasetSize(const std::vector<std::string>& input_data_paths)
```

Comment:
I don't like this function, as mentioned [here](https://github.com/triton-inference-server/perf_analyzer/pull/280/files#r1946939872), but it was a quick way to scan the input data JSON(s) and pre-check out many requests there will be so that that information can be given to InferenceProfiler class to determine when benchmarking is done.